lib/api_repo.js

```diff
@@ -572,59 +565,53 @@ var isProtoPath = /.proto$/;
  */
 ApiRepo.prototype._findProtos = function _findProtos(name, version, done,
                                                      onProto) {
-  // Determine the top-level dir, verify it exists, then scan for protos in it.
-  var parts = [this.repoDir];
+  // Determine the top-level proto dir
+  var topDir = this.repoDir;
+  var parts = [topDir];
   if (this.isGoogleApi) {
     parts.push('google');
   }
   parts.push(name, version);
   var protoDir = path.join.apply(path, parts);
-  var checkProtoDir = newIsDirFunc(protoDir);
-  var that = this;
+
+  /* Use glob to scan for protos in a protoDir */
   var scanForProtos = function scanForProtos(next) {
-    var finished = false,
-        protos = [];
-    var finder = new FindFiles({
-      rootFolder : protoDir,
-      filterFunction : function (path) {
-        return isProtoPath.test(path);
+    var stripRoot = function stripRoot(err, outputs) {
+      if (err) {
```

Comment:
The same pattern is seen above. Can we define a function like

    function processFile(done, proc) {
       return function(err, outputs) {
         if (err) {
           done(err);
         } else {
            var result = process(outputs);
            done(null, result);
         }
       }
    }

This way, the glob is simply

    glob.glob("*.proto", {
        ...
    }, processFiles(next, function(outputs) { return _.map(outputs, function(x) { ... }; })

Still a bit ugly, but not that bad (if arrow function is available, this is going to be `(outputs) => _.map(outputs, ...)`)